File: app/src/main/java/com/twismart/wallpapershd/WallpaperApplication.java
package com.twismart.wallpapershd;

import android.app.Application;

import com.crashlytics.android.Crashlytics;
import com.twismart.wallpapershd.data.DataManager;
import com.twismart.wallpapershd.di.component.ApplicationComponent;
import com.twismart.wallpapershd.di.component.DaggerApplicationComponent;
import com.twismart.wallpapershd.di.module.ApplicationModule;

import io.fabric.sdk.android.Fabric;
import javax.inject.Inject;

/**
 * Created by sneyd on 4/15/2017.
 **/

public class WallpaperApplication extends Application {

    @Inject
    DataManager mDataManager;

    private ApplicationComponent mApplicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        Fabric.with(this, new Crashlytics());
        initAppComponent();
    }

    public void initAppComponent(){
        mApplicationComponent = DaggerApplicationComponent.builder().applicationModule(new ApplicationModule(this)).build();
        mApplicationComponent.inject(this);
    }

    public ApplicationComponent getComponent() {
        return mApplicationComponent;
    }
}

File: app/src/main/kotlin/com/twismart/wallpapershd/WallpaperApplication.kt
package com.twismart.wallpapershd

import android.support.multidex.MultiDexApplication
import com.crashlytics.android.Crashlytics
import com.facebook.stetho.Stetho
import com.twismart.wallpapershd.di.component.ApplicationComponent
import com.twismart.wallpapershd.di.component.DaggerApplicationComponent
import com.twismart.wallpapershd.di.module.ApplicationModule
import io.fabric.sdk.android.Fabric

class WallpaperApplication : MultiDexApplication() {

    val component: ApplicationComponent by lazy {
         DaggerApplicationComponent
                .builder()
                .applicationModule(ApplicationModule(this))
                .build()
    }

    override fun onCreate() {
        super.onCreate()
        Fabric.with(this, Crashlytics())
        component.inject(this)
        Stetho.initializeWithDefaults(this)
    }
}